crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -0,0 +1,329 @@
+        pub extensions: BTreeMap<String, serde_json::Value>,
+
+    }
+
+    response: {
+        /// Present and true if this response describes an initial sync
+        /// (i.e. after the `pos` token has been discard by the server?)
+        #[serde(default, skip_serializing_if = "ruma_common::serde::is_default")]
+        pub initial: bool,
+
+        /// The token to supply in the `pos` param of the next `/sync` request.
+        pub pos: String,
+
+        /// Updates to the sliding room list.
+        #[serde(default = "Vec::new", skip_serializing_if = "Vec::is_empty")]
+        pub lists: Vec<SyncList>,
+
+        /// The updates on rooms.
+        #[serde(default = "BTreeMap::default", skip_serializing_if = "BTreeMap::is_empty")]
+        pub rooms: BTreeMap<OwnedRoomId, SlidingSyncRoom>
+    }
+
+    error: crate::Error
+}
+
+/// Filter for a sliding sync list, set at request.
+///
+/// All fields are applied with AND operators, hence if `is_dm`  is `true` and `is_encrypted` is
+/// `true` then only encrypted DM rooms will be returned. The absence of fields implies no filter
+/// on that criteria: it does NOT imply `false`.
+///
+/// Filters are considered _sticky_, meaning that the filter only has to be provided once and their
+/// parameters 'sticks' for future requests until a new filter overwrites them.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct SyncRequestListFilters {
+    // These fields may be expanded through use of extensions.
+    /// Whether to return DMs, non-DM rooms or both.
+    ///
+    /// Flag which only returns rooms present (or not) in the DM section of account data.
+    /// If unset, both DM rooms and non-DM rooms are returned. If false, only non-DM rooms
+    /// are returned. If true, only DM rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_dm: Option<bool>,
+
+    /// Only list rooms that are spaces of these or all
+    ///
+    /// A list of spaces which target rooms must be a part of. For every invited/joined
+    /// room for this user, ensure that there is a parent space event which is in this list. If
+    /// unset, all rooms are included. Servers MUST NOT navigate subspaces. It is up to the
+    /// client to give a complete list of spaces to navigate. Only rooms directly in these
+    /// spaces will be returned.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub spaces: Vec<String>,
+
+    /// Whether to return encrypted, non-encrypted rooms or both.
+    ///
+    /// Flag which only returns rooms which have an `m.room.encryption` state event. If
+    /// unset, both encrypted and unencrypted rooms are returned. If false, only unencrypted
+    /// rooms are returned. If true, only encrypted rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_encrypted: Option<bool>,
```

Comment:
To be clear: `serde(default)` isn't needed here because `Option` has it implicitly (although some other serde attributes seem to disable this default).